homeassistant/components/nest/event.py

```diff
@@ -0,0 +1,127 @@
+"""Event platform for Google Nest."""
+
+from dataclasses import dataclass
+import logging
+
+from google_nest_sdm.device import Device
+from google_nest_sdm.device_manager import DeviceManager
+from google_nest_sdm.event import EventMessage, EventType
+from google_nest_sdm.traits import TraitType
+
+from homeassistant.components.event import (
+    EventDeviceClass,
+    EventEntity,
+    EventEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DATA_DEVICE_MANAGER, DOMAIN
+from .device_info import NestDeviceInfo
+from .events import (
+    EVENT_CAMERA_MOTION,
+    EVENT_CAMERA_PERSON,
+    EVENT_CAMERA_SOUND,
+    EVENT_DOORBELL_CHIME,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class NestEventEntityDescription(EventEntityDescription):
+    """Entity description for nest event entities."""
+
+    trait_type: TraitType | None = None
+    api_event_type: EventType | None = None
+    event_type: str | None = None
```

Comment:
```suggestion
@dataclass(frozen=True, kw_only=True)
class NestEventEntityDescription(EventEntityDescription):
    """Entity description for nest event entities."""

    trait_type: TraitType
    api_event_type: EventType
    event_type: str
```

Reply:
I was getting mypy errors before due to default arguments, but maybe because i was missing `kw_only=True`?

Reply:
Yeah kw_only fixes it. Thanks.
```
  File "/opt/hostedtoolcache/Python/3.12.4/x64/lib/python3.12/dataclasses.py", line 1063, in _process_class
    _init_fn(all_init_fields,
  File "/opt/hostedtoolcache/Python/3.12.4/x64/lib/python3.12/dataclasses.py", line 585, in _init_fn
    raise TypeError(f'non-default argument {f.name!r} '
TypeError: non-default argument 'trait_types' follows default argument
```